src/savegame_json.cpp

```diff
@@ -2899,6 +2920,12 @@ void vehicle::serialize( JsonOut &json ) const
     json.member( "airworthy", flyable );
     json.member( "summon_time_limit", summon_time_limit );
     json.member( "magic", magic );
+
+    if( smart_controller_cfg ) {
+        json.member( "smart_controller" );
+        smart_controller_cfg -> serialize( json );
+    }
```

Comment:
```suggestion
    json.member( "smart_controller", smart_controller_cfg );
```
`cata::optional` can be directly written with `JsonOut` when the value it contains supports the `serialize` method.